src/driver/franka_plan_runner.cc

```diff
@@ -193,17 +190,13 @@ int FrankaPlanRunner::RunFranka() {
             fmt::format("robot cannot receive commands in mode: {} at startup",
                         utils::RobotModeToString(current_mode))};
         comm_interface_->PublishDriverStatus(false, err_msg);
-        if (t_now - t_last_main_loop_log_ >= std::chrono::seconds(1)) {
-          dexai::log()->error("RunFranka: {}", err_msg);
-          t_last_main_loop_log_ = t_now;
+        if (current_mode == franka::RobotMode::kUserStopped) {
+          comm_interface_->PublishBoolToChannel(
+              utils::get_current_utime(),
+              comm_interface_->GetUserStopChannelName(), true);
```

Comment:
move handling for `franka::RobotMode::kUserStopped` mode before L188 `else if`, as opposed to inside `else if (current_mode != franka::RobotMode::kIdle) {`

Reply:
I don't see why this is necessary. This is simpler and cleaner

Reply:
Avoid duplicating time-checking logic and printout

Reply:
that section is structured to handle known individual cases first in some order, and then handle everything `else`. you cannot put a special case under the last `else` which breaks the code structure